packages/@lwc/engine-core/src/framework/register-stylesheets.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { ArrayPush, defineProperty, isUndefined } from '@lwc/shared';
+import { Template } from './template';
+import { TemplateStylesheetFactories } from './stylesheet';
+import { flattenStylesheets } from './utils';
+import { checkVersionMismatch } from './check-version-mismatch';
+
+export function registerStylesheets(
+    tmpl: Template,
+    stylesheetToken: string,
+    ...stylesheetLists: Array<TemplateStylesheetFactories | undefined>
+) {
+    tmpl.stylesheetToken = stylesheetToken;
+    tmpl.stylesheets = [];
+    for (const stylesheets of stylesheetLists) {
```

Comment:
For a follow-up PR, if we were to have a method that has access to both the template and the stylesheet, we do run the stylesheet flattening here. The template type could look like this:

```diff
export interface Template {
-    stylesheets?: TemplateStylesheetFactories
+    stylesheets?: StylesheetFactory;
}
```